app-modules/engagement/src/Notifications/EngagementFailedNotification.php

```diff
@@ -69,10 +70,18 @@ public function toEmail(object $notifiable): MailMessage
 
     public function toDatabase(object $notifiable): array
     {
+        $engagementType = '';
+
+        if ($this->engagement->deliverable->channel == EngagementDeliveryMethod::Email) {
+            $engagementType = $this->engagement->deliverable->channe;
+        } elseif ($this->engagement->deliverable->channel == EngagementDeliveryMethod::Sms) {
+            $engagementType = 'SMS';
+        }
```

Comment:
A `match` statement would probably work better here.

Also the results for the email type seems incomplete.

Reply:
@Orrison Made the changes